e2e_test/bin/e2e_test.dart

```diff
@@ -51,11 +78,12 @@ Future<bool> _waitForEventToShowUp(Uri url) async {
     );
     print('${response.statusCode}: ${response.body}');
     if (response.statusCode == 200) {
-      return true;
+      // The json does not match what `SentryEvent.fromJson` expects
+      return jsonDecode(utf8.decode(response.bodyBytes));
```

Comment:
how so? it should, can you send me the payload?

Reply:
```json
{"event_id":"1a83e0da67914827909cc8dee3ca28d9","project":5428562,"release":null,"dist":null,"platform":"other","message":"","datetime":"2021-08-31T09:57:24.247000Z","tags":[["environment","e2e"],["level","error"],["os","linux Linux 5.8.0-1039-azure #42~20.04.1-Ubuntu SMP Thu Jul 15 14:11:07 UTC 2021"],["os.name","linux"]],"_metrics":{"bytes.ingested.event":1150,"bytes.stored.event":2226},"contexts":{"dart_context":{"number_of_processors":2,"package_config":"file:///home/runner/work/sentry-dart/sentry-dart/e2e_test/.dart_tool/package_config.json","type":"dart_context"},"device":{"name":"fv-az399-758","timezone":"UTC","language":"C.UTF-8","type":"device"},"os":{"name":"linux","version":"Linux 5.8.0-1039-azure #42~20.04.1-Ubuntu SMP Thu Jul 15 14:11:07 UTC 2021","type":"os"},"sentry_dart_runtime":{"name":"Dart","raw_description":"2.15.0-65.0.dev (dev) (Fri Aug 27 11:53:11 2021 -0700) on \"linux_x64\"","type":"sentry_dart_runtime"}},"culprit":"e2e_test.dart in main","environment":"e2e","exception":{"values":[{"type":"_Exception","value":"Exception: E2E Test Message","stacktrace":{"frames":[{"function":"main","filename":"e2e_test.dart","abs_path":"e2e_test.dart","lineno":27,"colno":5,"in_app":true}]}}]},"fingerprint":["{{ default }}"],"grouping_config":{"enhancements":"eJybzDRxY3J-bm5-npWRgaGlroGxrpHxpEkTNxXkF6XrpiQWleQk5qXrFqdka008y8yIS4IBANd-Gx4","id":"newstyle:2019-10-29"},"hashes":["f470a3c7079f68d620bd4204ac51ff1a"],"ingest_path":[{"public_key":"XE7QiyuNlja9PZ7I9qJlwQotzecWrUIN91BAO7Q5R38","version":"21.8.0"}],"key_id":"1764334","level":"error","location":"e2e_test.dart","logger":"","metadata":{"filename":"e2e_test.dart","function":"main","type":"_Exception","value":"Exception: E2E Test Message"},"received":1630403844.498328,"sdk":{"name":"sentry.dart","version":"6.0.0-beta.5","integrations":["isolateErrorIntegration"],"packages":[{"name":"pub:sentry","version":"6.0.0-beta.5"}]},"timestamp":1630403844.247,"title":"_Exception: Exception: E2E Test Message","type":"error","version":"7"}
```

from https://sentry.io/share/issue/b378131b171643b28ebd70c8f0c430a6/

Reply:
So for example tags in the event (our event and docs) look like
```json
{
  "tags": {
    "ios_version": "4.0",
    "context": "production"
  }
}
```
but actually look like
```json
{
"tags":[
      [
         "environment",
         "e2e"
      ],
      [
         "level",
         "error"
      ],
      [
         "os",
         "linux Linux 5.8.0-1039-azure #42~20.04.1-Ubuntu SMP Thu Jul 15 14:11:07 UTC 2021"
      ],
      [
         "os.name",
         "linux"
      ]
   ],
}
```

Reply:
It is also kinda documented that they look kinda different: https://docs.sentry.io/api/events/retrieve-an-event-for-a-project/

Reply:
oh got it, indeed, as it returns a Map, we could assert by key name then following the contract described above, TIL